target_chains/cosmwasm/pyth-sdk-cw/src/testing.rs

```diff
@@ -0,0 +1,87 @@
+use {
+    crate::{
+        error::PythContractError,
+        PriceFeed,
+        PriceFeedResponse,
+        PriceIdentifier,
+        QueryMsg,
+    },
+    cosmwasm_std::{
+        from_binary,
+        to_binary,
+        Binary,
+        Coin,
+        ContractResult,
+        QuerierResult,
+        SystemError,
+        SystemResult,
+    },
+    std::{
+        collections::HashMap,
+        time::Duration,
+        u128,
+    },
+};
+
+/// Mock version of Pyth for testing cosmwasm contracts.
+/// This mock stores some price feeds and responds to query messages.
+#[derive(Clone)]
+pub struct MockPyth {
+    pub valid_time_period: Duration,
+    pub fee_per_vaa:       Coin,
+    pub feeds:             HashMap<PriceIdentifier, PriceFeed>,
+}
+
+impl MockPyth {
+    /// Create a new `MockPyth`. You can either provide the full list of price feeds up front,
+    /// or add them later via `add_feed`.
+    pub fn new(valid_time_period: Duration, fee_per_vaa: Coin, feeds: &[PriceFeed]) -> Self {
+        let mut feeds_map = HashMap::new();
+        for feed in feeds {
+            feeds_map.insert(feed.id, *feed);
+        }
+
+        MockPyth {
+            valid_time_period,
+            fee_per_vaa,
+            feeds: feeds_map,
+        }
+    }
+
+    /// Add a price feed that will be returned on queries.
+    pub fn add_feed(&mut self, feed: PriceFeed) {
+        self.feeds.insert(feed.id, feed);
+    }
+
+    /// TODO: Update this with example contracts -> Handler for processing query messages. See the tests in `contract.rs` for how to use this
+    /// handler within your tests.
```

Comment:
i note that contract.rs isn't included here at the moment. if you're going to add that in a future PR, great. If not, we should document how to use this somewhere.

Reply:
Yes, the example contract here - https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/cosmwasm/examples/cw-contract/src/contract.rs 

Will update the example in future PRs. This Todo will be resolved then.